libcaf_test/caf/test/runnable.hpp

```diff
@@ -199,6 +187,14 @@ class CAF_TEST_EXPORT runnable {
       rep->test_stats({before.passed + 1, before.failed});
   }
 
+  template <class T0, class T1>
+  static void assert_save_comparison(T0&&, T1&&) {
```

Comment:
As I said before, this function does not need actual function arguments:

```suggestion
  static void assert_save_comparison() {
```

Reply:
We can not use this approach as the compiler will not be able to deduce the types in this case. This would work if the whole class was templated.

Reply:
Of course the deduction won't work if there aren't arguments. You need to call it differently: `assert_save_comparison<T0, T1>()`.

Reply:
Sorry, I did not understand that you meant to use the classes as template argument directly. I have updated the code accoordingly.